examples/untyped-py/step1/__main__.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2021, Pulumi Corporation.  All rights reserved.
+
+import pulumi
+from pulumi_aws_native import ExtensionResource, Provider, ProviderDefaultTagsArgs
+from pulumi_random import RandomId
+
+aws_native_config = pulumi.Config("aws-native")
+
+aws_native_provider = Provider("aws_native_provider",
+                               region=aws_native_config.require("region"),
+                               default_tags=ProviderDefaultTagsArgs(tags={"defaultTag": "defaultTagValue"}))
+
+paramName = RandomId("paramName", byte_length=8)
+
+resource = ExtensionResource("param",
+                             type="AWS::SSM::Parameter",
+                             properties={
+                                 "Name": paramName.id,
+                                 "Type": "String",
+                                 "Value": "originalValue",
+                                 "Tags": {
+                                     "localTag": "localTagValue",
+                                 }
+                             },
+                             write_only=[
+                                 "AllowedPattern",
+                                 "Description",
+                                 "Policies",
+                                 "Tags",
+                                 "Tier"
+                             ],
+                             tags_property="Tags",
+                             tags_style="stringMap",
```

Comment:
We need to make them replaceOnChanges, and the typeName too.

Reply:
Yes, it's not ideal that users have to provide these, though they are optional so it can be used without them (though write-only properties can result in updates failing if not specified).

To make the most common patterns work without user input, perhaps we should change the default `tagsProperty` to be `Tags` if unspecified (which would be the case for >90% of resources) and default the `tagsStyle` to `keyValueArray` which is also the most common style. In the less common case of wanting to opt-out of default tags the user could then set `tagsStyle` to `none`.

Reply:
On second thoughts, I think defaulting to blindly adding a tags field is probably surprising behaviour as this could lead to the extension resource payload being invalid - adding a "Tags" field to something that doesn't expect tags. Therefore, I think we need some kind of basic opt-in for applying tags which I'd suggest should be by specifying either the `tagsProperty` or `tagsStyle` - we will the apply the default to the other.

Reply:
I've added the `replaceOnChanges`, however, due to https://github.com/pulumi/pulumi/issues/15726 this has no effect since we removed all inputs from also being outputs. The gist of the bug is that we actually mark which inputs cause replacements on the output property 🙃 

At some point when the bug it fixed and we upgrade our dependency this will resolve itself.